IngilevichVarvara/cafeMenu.js

```diff
@@ -0,0 +1,250 @@
+        } else {
+            throw new Error("You have not specified the information about the size of hamburger!");
+        }
+    } else {
+        MenuItem.call(this);
+        this.size = size.name;
+        this.stuffing = stuffing.name;
+        this.price = size.price + stuffing.price;
+        this.calories = size.calories + stuffing.calories;
+    }
+
+}
+
+//Methods inheritance
+Hamburger.prototype = Object.create(MenuItem.prototype);
+
+/**
+ * Узнать размер гамбургера
+ */
+Hamburger.prototype.getSize = function () {
+    return this.size;
+};
+
+/**
+ * Узнать начинку гамбургера
+ */
+Hamburger.prototype.getStuffing = function () {
+    return this.stuffing;
+};
+
+
+//Salads
+const ONE_UNIT_OF_SALAD_IN_GRAM = 100;
+
+function Salad(salad, grams) {
+    if (arguments.length > 2) {
+        throw new Error("You have provided too much information about the salad. Please specify salad and its weight in grams only.");
+    } else if (arguments.length === 0) {
+        throw new Error("You have not specified salad and its weight in grams.");
+    } else if (arguments.length === 1) {
+        if (typeof Array.prototype.slice.call(arguments)[0] === "object") {
+            throw new Error("You have not specified the weight of the salad!")
+        } else {
+            throw new Error("You have not specified the salad!")
+        }
+    } else {
+        MenuItem.call(this);
+        this.price = (salad.price * grams) / ONE_UNIT_OF_SALAD_IN_GRAM;
+        this.calories = salad.calories;
+        this.weight = grams;
+        this.name = salad.name;
+    }
+}
+
+Salad.prototype = Object.create(MenuItem.prototype);
+
+/**
+ * Узнать вес салата
+ */
+Salad.prototype.getWeight = function () {
+    return this.weight;
+};
+
+/**
+ * Узнать название салата
+ */
+Salad.prototype.getSaladName = function () {
+    return this.name;
+};
+
+//Drinks
+function Drink(drink) {
+    if (arguments.length === 0) {
+        throw new Error("You have not provided any information about the drink!");
+    } else {
+        MenuItem.call(this);
+        this.price = drink.price;
+        this.calories = drink.calories;
+        this.name = drink.name;
+    }
+}
+
+Drink.prototype = Object.create(MenuItem.prototype);
+
+/**
+ * Узнать название напитка
+ */
+Drink.prototype.getDrinkName = function () {
+    return this.name;
+};
+
+function Order() {
+    this.orderItems = Array.prototype.slice.call(arguments);
+}
+
+/**
+ * Узнать цену заказа
+ */
+Order.prototype.getOrderPrice = function () {
+    return this.orderItems.reduce(function (sum, currentItem) {
+        return sum + currentItem.calculatePrice()
+    }, 0);
+};
+
+/**
+ * Узнать калории заказа
+ */
+Order.prototype.getOrderCalories = function () {
+    return this.orderItems.reduce(function (sum, currentItem) {
+        return sum + currentItem.calculateCalories()
+    }, 0);
+};
+
+/**
+ * Добавить позицию в заказ
+ */
+Order.prototype.addItem = function (item) {
```

Comment:
Maybe it would convenient to change this function signature to accept the array of items. In this case, you would be able to add several items to the order:

```
Order.prototype.addItems = function (...items) {
  if (items.length > 0) {
    items.forEach(/* ... */)
  }
}
```